### PR TITLE
Handle item attachments with missing name

### PR DESCRIPTION
**Pull request by pandeyabs**
Extending the earlier fix in https://github.com/alcionai/corso/pull/5199 to `itemAttachments`. Posts are not impacted here since they don't have attachment types like messages do.

---

#### Does this PR need a docs update or release note?

- [x] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [ ] :no_entry: No

#### Type of change

- [ ] :sunflower: Feature
- [x] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

#### Issue(s)

* #<issue>

#### Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: src/internal/converters/eml/eml_test.go [hunk 1]
```diff
@@ -461,7 +468,7 @@ func (suite *EMLUnitSuite) TestConvert_message_in_messageble_to_eml() {
 	assert.Equal(t, formatAddress(msg.GetFrom().GetEmailAddress()), eml.GetHeader("From"))
 
 	attachments := eml.Attachments
-	assert.Equal(t, 1, len(attachments), "attachment count in parent email")
+	assert.Equal(t, 3, len(attachments), "attachment count in parent email")
```
Comment (vkamra) [inline src/internal/converters/eml/eml_test.go:471]: Having a hard time finding which test case in testdata covers the bug fix here? Can you add a comment in the test data file?

**Review by vkamra — APPROVED** [review]
Change LGTM.

File: src/internal/converters/eml/eml_test.go (hunk 1 above)
Comment (pandeyabs) [reply src/internal/converters/eml/eml_test.go:471]: The test cases are at line 142(`TestConvert_edge_cases`). Basically the same tests we had for file attachments, we are running them now for item attachments as well. The testdata has 3 identical item attachments, with diff IDs. We `transform` the json testdata in `TestConvert_edge_cases` to overwrite attachment names for these test cases. 

I'll add a comment.

Comment (vkamra) [reply src/internal/converters/eml/eml_test.go:471]: Thanks. Does this include a case where the attachment doesn't have a name?

Comment (pandeyabs) [reply src/internal/converters/eml/eml_test.go:471]: Yep. Here https://github.com/alcionai/corso/pull/5209/files#diff-1932db8354803639e32f2bbaf0fbfba246fb2b2e3a6490f02a8e980d8413e003R171

**Pull request merged**